### PR TITLE
do not start nginx until /var/run/php-fpm.sock is available

### LINKED ISSUE #42: The docker image starts nginx before php-fpm leading to 502 gateway not found
## Steps to reproduce
1. Deploy PrivateBin to Google Cloud Run
2. Run an intense load test with 50 concurrent users
```shell-session
$ go install github.com/k6io/xk6/cmd/xk6@latest
$ xk6 build --with github.com/binxio/xk6-privatebin@v0.1.0
$ cat > test.js <<!
import http from 'k6/http';
import {
    sleep,
    check
} from 'k6
import privatebin from 'k6/x/privatebin';


export default function() {
    var params = {
        headers: {
            'X-Requested-With': 'JSONHttpRequest',
        },
    };
    var paste = privatebin.encrypt("hello world!");

    var result = http.post('https://privatebin.net', paste.body, params);
    if (check(result, {
            'post status 200': (r) => r.status === 200,
            'post ok': (r) => r.headers['Content-Type'].startsWith('application/json') && r.json().status === 0,
        })) {
        var url = `${result.url}?${result.json().id}`;
        var result = http.get(url, params);
        if (!check(result, {
                'get status 200': (r) => r.status === 200,
                'get data equal': (r) => r.headers['Content-Type'].startsWith('application/json') && JSON.stringify(r.json().adata) === JSON.stringify(JSON.parse(paste.body).adata)
            })) {
            console.log(result.url);
            console.log(result.body);
        }
    } else {
        console.log(JSON.stringify(result));
    }
    sleep(0.5);
}
!
$ ./k6  run -u 50 -i 1000 test.js
```

### What happens
The first calls to PrivateBin will return a 502, as nginx is accepting the http requests but php-fpm has not completed the initialization and /var/run/php-fpm.sock is not available.

```html
<html>
   <head><title>502 Bad Gateway</title></head>
   <body>
       <center><h1>502 Bad Gateway</h1></center>
       <hr>
       <center>nginx</center>
   </body>
</html>
```

Once php-fpm is up, privatebin starts functioning properly.

## Additional information

I added the following script to the container:

```shell
#!/bin/sh
test -S /var/run/php-fpm.sock || (echo "waiting on /var/run/php-fpm.sock" >&2 && sleep 1 && exit 1)
```

And changed `/etc/s6/services/nginx/run` to include an if statement:

```
#!/usr/bin/execlineb -P
if { /usr/local/bin/php-fpm-sock-is-available }
/usr/sbin/nginx
```

## Basic information

**Server OS:** alpine:3.13

**Webserver:** nginx installed with alpine 3.13
 

**PrivateBin version:** 1.3.5

I *cannot* reproduce this issue on <https://privatebin.net> as this only occurs on the bootstrap of PrivateBin (easily reproducible on Google Cloud Run).

### PR DESCRIPTION
**Pull request by mvanholsteijn**
closes #42

# Changes
- add test for existence of the pipe /var/run/php-fpm.sock

### CODE REVIEW

File: etc/s6/services/nginx/run [hunk 1]
```diff
@@ -1,2 +1,3 @@
 #!/usr/bin/execlineb -P
+if { test -S /var/run/php-fpm.sock }
```
Comment (rugk) [inline etc/s6/services/nginx/run:2]: Is this really the full syntax?
In https://github.com/PrivateBin/docker-nginx-fpm-alpine/issues/42 you said you added some shell script instead?

**Comment by elrido** [comment]
Thanks for submitting this!

File: etc/s6/services/nginx/run (hunk 1 above)
Comment (elrido) [reply etc/s6/services/nginx/run:2]: Yeah, I'll extend the scriptlet `test -S /var/run/php-fpm.sock || (echo "waiting on /var/run/php-fpm.sock" >&2 && sleep 1 && exit 1)` from there a little bit. It should be possible to wait in a while loop (ideally with no delay if the socket already exists), but time out after i.e. 15 seconds if the php process never came up, so the container run gets failed and the admin can handle it.

Comment (mvanholsteijn) [reply etc/s6/services/nginx/run:2]: I added the sleep script, to implement a sleep; until I found out that s6 will retry every second when then run script returns an error. 

When the socket does not appear, I trust on php-fpm producing appropriate errors.

**Pull request merged**